packages/sdk/src/server/index.ts

```diff
@@ -353,7 +353,10 @@ export class BackendClient {
    * @param opts - The options for configuring the server client.
    */
   constructor(opts: BackendClientOpts) {
-    this.environment = opts.environment ?? "production";
+    if (!opts.environment || !["development", "production"].includes(opts.environment)) {
```

Comment:
Small but I'd suggest moving this check to its own private method since we may want to validate this elsewhere in the SDK, and since the env validation logic may be more complex later